packages/xdl/src/ApiV2.ts

```diff
@@ -163,6 +187,8 @@ export default class ApiV2Client {
     }
 
     reqOptions = merge({}, reqOptions, extraRequestOptions);
```

Comment:
```suggestion
    reqOptions = merge({}, reqOptions, extraRequestOptions, uploadOptions);
```

then remove the next line. see https://lodash.com/docs/#merge for more info and https://snack.expo.io/@notbrent/joyous-popcorn for an example